topology/src/main/java/io/camunda/zeebe/topology/state/TopologyChangeOperation.java

```diff
@@ -8,12 +8,35 @@
 package io.camunda.zeebe.topology.state;
 
 import io.atomix.cluster.MemberId;
+import java.util.Optional;
 
 /**
  * An operation that changes the topology. The operation could be a member join or leave a cluster,
  * or a member join or leave partition.
  */
-public interface TopologyChangeOperation {
+public record TopologyChangeOperation(MemberId memberId, Operation operation) {
 
-  MemberId memberId();
+  public record PartitionOperation(
+      int partitionId, PartitionOperationType operationType, Optional<Integer> priority)
+      implements Operation {
+    @Override
+    public boolean isPartitionOperation() {
+      return true;
+    }
+  }
+
+  public interface Operation {
+    Operation NONE = new Operation() {};
+
+    default boolean isPartitionOperation() {
+      return false;
+    }
+  }
+
+  public enum PartitionOperationType {
+    UNKNOWN,
```

Comment:
It is not required here. I think it was from the memory of protobuf, where the recommendation is to always define an unknown. It is not required here, because this is not directly serialized. Will remove.